### PR TITLE
gdbm: update patch to match upstream

### PR DESCRIPTION
**Pull request by FnControlOption**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

My original patch [doesn't work on Linux](https://github.com/Homebrew/discussions/discussions/2184) for some reason, so I've updated the formula to use the upstream fix.

I updated the `configure` script by manually running `autoconf`, etc.

### CODE REVIEW

File: Formula/gdbm.rb [hunk 1]
```diff
@@ -50,20 +51,129 @@ def install
 end
 
 __END__
---- src/gdbmshell.c.orig
+commit 32517af75ac8c32b3ff4870e14ff28418696c554
```
Comment (carlocab) [inline Formula/gdbm.rb:54]: Thanks, @FnControlOption. Do you mind submitting this patch to Homebrew/formula-patches?

Comment (FnControlOption) [reply Formula/gdbm.rb:54]: https://github.com/Homebrew/formula-patches/pull/396

**Comment by iMichka** [comment]
I fixed the patch manually. Our merge process to linuxbrew-core is a little bit ... broken regarding patches, and sometimes tabs/spaces get moved away because my text editor is cleaning these up. A bottle has been built in linuxbrew-core so the current status is that everything should be fine.

But I do agree that moving the patch to formula-patches is a better practice.

**Review by carlocab — APPROVED** [review]
Thanks @FnControlOption!

**Pull request closed without merging**